### ISSUE TITLE
Low accuracy on HMDB51

### ISSUE THREAD

**Issue by vateye** [kenshohara/3D-ResNets-PyTorch#49]
Hi, nice work first.
I have finetune my model on hmdb51, then I got a checkpoint save_200.pth. Then I try to run the following script to evaluate on validation.

`python main.py --root_path ~/Research/datasets --video_path hmdb51/jpg --annotation_path hmdb51/testTrainMulti/hmdb51_1.json --result_path ~/Research/3D-ResNets-PyTorch/results/hmdb51 --dataset hmdb51 --n_finetune_classes 51 --n_classes 51 --model resnext --model_depth 101 --resnet_shortcut B --resnext_cardinality 32 --batch_size 32 --n_threads 4 --test --test_subset val --pretrain_path ~/Research/3D-ResNets-PyTorch/results/hmdb51/save_200.pth --no_train --no_val`

After that I got a file called val.json. Then I run evaluate the script you provide in utils/eval_hmdb51.py.

`hmdb = HMDBclassification(ground_truth_filename='hmdb51_1.json',prediction_filename='val.json',verbose=True, top_k=1)
hmdb.evaluate()
`
Then I got 
`[INIT] Loaded annotations from validation subset.
	Number of ground truth instances: 1530
	Number of predictions: 15290
[RESULTS] Performance on ActivityNet untrimmed video classification task.
	Error@1: 0.9784313725490196`

It seems that something wrong with the prediction numbers. Can you tell me how you run the evaluation script. Thx :)

**Comment by Minju-Jung** [comment]
I got the same problem.
I found that "model.py" ignores the final pre-trained fc layer by making a new fc layer.

In "model.py", I commented out the following lines.
`        if opt.model == 'densenet':
            model.module.classifier = nn.Linear(
                model.module.classifier.in_features, opt.n_finetune_classes)
            if not opt.no_cuda:
                model.module.classifier = model.module.classifier.cuda()
        else:
            model.module.fc = nn.Linear(model.module.fc.in_features,
                                        opt.n_finetune_classes)
            if not opt.no_cuda:
                model.module.fc = model.module.fc.cuda()`

**Comment by vateye** [comment]
@Minju-Jung I think it's not the new fc layer's fault, because our goal is to finetune on other datasets. Making a new fc layer is reasonable. I think maybe something wrong in "test.py" ?

**Comment by Minju-Jung** [comment]
When you start to fine-tune the network for HMDB51, a new fc layer allocation is correct.
However, what I want to say is that a new fc layer allocation should not be occurred when you load the trained network.
I think that the actual network you used for testing cannot load the final fc layer parameters in  ~/Research/3D-ResNets-PyTorch/results/hmdb51/save_200.pth.

**Comment by vateye** [comment]
@Minju-Jung I think I'm fully understanding what you means. Before I do testing, I should comment out the above line, then I use my script to test.
But what about the problem below ?

> Number of ground truth instances: 1530
> Number of predictions: 15290

I think the  'Number of predictions' should equal to 'Number of ground truth instances * 10' ?

**Comment by vateye** [comment]
@kenshohara Hi, I have a question, in paper the HMDB's accuracy is 63.8%. But I've tried on split1, for the validation set the accuracy achieve's 64.7%, but on test set, it only be 53.8%. Any suggestion?

**Comment by kenshohara** [comment]
@vateye 
What is the test set of HMDB51?
HMDB51 only includes training set and test set,
and I named test set validation set in my json file.

**Comment by vateye** [comment]
@kenshohara Sorry, I use the video which is not in the training and test set as the test set.
And I’m still not clear about the during testing, the number of predictions is not equal to the number of ground truth times 10.

**Comment by kenshohara** [comment]
I have not confirmed the videos of test set,
so I cannot answer the performance difference between the validation and test set.

In my environment, the number of predictions is also 15290.
I think one video was not loaded in the test step by some reasons,
but it did not influence the accuracy significantly because it is about only one video.
